src/autohost.ts

```diff
@@ -325,6 +326,43 @@ export class Autohost implements TachyonAutohost {
 			);
 		}
 	}
+
+	shutdown(): void {
```

Comment:
I don't believe this logic belongs in this class, `Autohost` implements the functionality as required by tachyon protocol, it focused on bridging the gap between tachyon protocol and engine protocol classes, and not full application lifetime management.

The orchestration of shutdown IMHO should be contained somewhere around the `main.ts`, *maybe* it's time to introduce `Application` class for example if it would help with testability of some of that etc. I don't have full sketch of implementation, but I have clear idea what should be responsibility of each piece so it's clear.